applications/stepswitch/src/stepswitch_util.erl

```diff
@@ -49,11 +50,48 @@ get_realm(JObj) ->
 -spec get_inbound_destination(kz_json:object()) -> kz_term:ne_binary().
 get_inbound_destination(JObj) ->
     {Number, _} = kapps_util:get_destination(JObj, ?APP_NAME, <<"inbound_user_field">>),
-    case kapps_config:get_is_true(?SS_CONFIG_CAT, <<"assume_inbound_e164">>, 'false') of
+    convert_to_e164_format(Number, ?SS_CONFIG_CAT, <<"assume_inbound_e164">>).
+
+%%------------------------------------------------------------------------------
+%% @doc Convert number to e164 format depending on config values set in system_config db
+%% @end
+%%------------------------------------------------------------------------------
+-spec convert_to_e164_format(kz_term:ne_binary(), kz_term:ne_binary(), kz_term:ne_binary()) -> kz_term:ne_binary().
+convert_to_e164_format(Number, ConfigDoc, ConfigKey) ->
+    case kapps_config:get_is_true(ConfigDoc, ConfigKey, 'false') of
         'true' -> assume_e164(Number);
         'false' -> knm_converters:normalize(Number)
     end.
 
+%%------------------------------------------------------------------------------
+%% @doc Get the origination DID of a inbound call if its a number, else return undefined
+%% @end
+%%------------------------------------------------------------------------------
+-spec get_inbound_origination(kz_json:object()) -> kz_term:ne_binary().
+get_inbound_origination(JObj) ->
+    {Number, _} = kapps_util:get_origination(JObj, ?APP_NAME, <<"inbound_user_origination_field">>),
+    case verify_is_did(Number) of
```

Comment:
@bradfordben would be better to use:

knm_converters:is_reconcilable(Number)

instead of:

verify_is_did/1

The knm_converters:is_reconcilable/1 is oddly named but basically answers the question "is this string a number that could be routed outside kazoo (IE: it is NOT an internal extension, or is in the format of a public number)?".  Since that is under the system admin's control it would match internationalization rules rather than just attempting to conver any integer.  Put another way, the logic would become "if the inbound user origination field username (number) is not an internal kazoo extension then attempt to convert it to e164".

Reply:
Great thanks, sounds good. Il make that change now and resubmit the PR.